src/main/java/org/opentripplanner/graph_builder/module/DirectTransferGenerator.java

```diff
@@ -133,6 +129,34 @@ public void buildGraph(
         graph.hasDirectTransfers = true;
     }
 
+    /**
+     * Add the computed transfer to the list of distinct ones for the two stops. If the identical
+     * one has already been computed for another mode (for example when the walking and cycling
+     * transfers use the same edges) then the new mode is added to the existing transfer.
+     */
+    private void updateDistinctTransfers(
+            Map<TransferKey, PathTransfer> distinctTransfers,
+            Stop stop,
+            StreetMode mode,
+            NearbyStop sd
+    ) {
+        var key = new TransferKey(stop, sd.stop, sd.edges);
+        if (distinctTransfers.containsKey(key)) {
+            var existingIdenticalTransfer = distinctTransfers.get(key);
+            var updatedTransfer = existingIdenticalTransfer.copyWithAddedMode(mode);
```

Comment:
Do we need to create a new transfer, or can we just mutate the modes set?

Reply:
Strictly speaking we can mutate but I'm trying to make the graph less mutable and hence prefer to create immutable classes. However, that might just be the Scala developer in me.

Reply:
We can discuss this tomorrow.